Issues with Elasticssearch test builds

By runing elasticsearch test builds we found there are some issues related with automatic running

- [x] 1. https://github.com/AdoptOpenJDK/openjdk-tests/pull/1087

 If there are errors tests will stop. [Java Testing Framework](https://www.elastic.co/guide/en/elasticsearch/reference/current/testing-framework.html)  only supports running all tests without stopping on failures by option [-Dtests.haltonfailure=false](https://github.com/elastic/elasticsearch/blob/master/TESTING.asciidoc) . The underlying framework is Junit so that may be the main reason. 

```
21:51:49     [junit4] JVM J3:     1.83 ..    12.11 =    10.28s
21:51:49     [junit4] JVM J4:     1.82 ..    23.67 =    21.85s
21:51:49     [junit4] JVM J5:     1.88 ..    26.94 =    25.06s
21:51:49     [junit4] JVM J6:     1.85 ..    24.17 =    22.32s
21:51:49     [junit4] JVM J7:     1.81 ..    19.91 =    18.10s
21:51:49     [junit4] Execution time total: 34 seconds
21:51:49     [junit4] Tests summary: 10 suites, 55 tests, 1 error
21:51:49  
21:51:49  
21:51:49  FAILURE: Build failed with an exception.
```

https://ci.adoptopenjdk.net/view/Test_external/job/openjdk11_hs_external_extended_elasticsearch_x86-64_linux/10/
Stop with:

```
21:51:49  BUILD FAILED in 28m 5s
21:51:49  223 actionable tasks: 114 executed, 109 up-to-date
```

https://ci.adoptopenjdk.net/view/Test_external/job/openjdk11_hs_external_extended_elasticsearch_x86-64_linux/9/
Successful with:

```
22:33:58  BUILD SUCCESSFUL in 1h 28m 3s
22:33:58  589 actionable tasks: 343 executed, 246 up-to-date
```

- [x] 2. https://github.com/AdoptOpenJDK/openjdk-tests/pull/1055

 We are currently running test from the root directory, which is telling gradle to run all test tasks. If we'd like to run a single test task we can either cd into the subproject where the test exists, or use the explicit gradle task like :server:test and with option of 'Test case filtering'. However it doesn't have an option to skip a test task|test class| test method unless tests code is with certain annotations.  

- [x] 3. https://github.com/AdoptOpenJDK/openjdk-tests/issues/1057

 The current junit test report strategy is running the tests locally to find out where the junit reports are generated, and then use that path in the docker cp command to copy to host machine. This is kind of hardcoded( same for other thirdparty tests), which only works for certain version. If anything updates some reports may be missing. See the error in jdk11 tests, build successfully run all the test.

```
22:33:58  BUILD SUCCESSFUL in 1h 28m 3s
22:33:58  589 actionable tasks: 343 executed, 246 up-to-date
22:33:59  
22:33:59  elasticsearch_test_0_PASSED
22:33:59  
22:33:59  Error: No such container:path: elasticsearch-test:/elasticsearch/core/build/reports/testJunit
22:33:59  Error: No such container:path: elasticsearch-test:/elasticsearch/modules/transport-netty4/build/reports/testJunit
22:33:59  Error: No such container:path: elasticsearch-test:/elasticsearch/modules/lang-mustache/build/reports/testJunit
22:34:00  Error: No such container:path: elasticsearch-test:/elasticsearch/modules/percolator/build/reports/testJunit
22:34:00  Error: No such container:path: elasticsearch-test:/elasticsearch/modules/reindex/build/reports/testJunit
22:34:00  Error: No such container:path: elasticsearch-test:/elasticsearch/modules/parent-join/build/reports/testJunit
22:34:00  Error: No such container:path: elasticsearch-test:/elasticsearch/test/framework/build/reports/testJunit
22:34:00  Error: No such container:path: elasticsearch-test:/elasticsearch/test/logger-usage/build/reports/testJunit
22:34:00  Error: No such container:path: elasticsearch-test:/elasticsearch/client/rest/build/reports/testJunit
22:34:11  elasticsearch-test
```

https://ci.adoptopenjdk.net/view/Test_external/job/openjdk11_hs_external_extended_elasticsearch_x86-64_linux/9/

https://ci.adoptopenjdk.net/view/Test_external/job/openjdk8_j9_external_sanity_payara-mp-tck_x86-64_linux/10/console

--- a/thirdparty_containers/elasticsearch/dockerfile/elasticsearch-test.sh
+++ b/thirdparty_containers/elasticsearch/dockerfile/elasticsearch-test.sh
@@ -20,7 +20,7 @@ if [ -d /java/jre/bin ];then
 	export JAVA_HOME=/java
 	export PATH=$JAVA_BIN:$PATH
 elif [ -d /java/bin ]; then
-	echo "Using mounted Java9"
+	echo "Using mounted Java"
 	export JAVA_BIN=/java/bin
 	export JAVA_HOME=/java
 	export PATH=$JAVA_BIN:$PATH
@@ -41,13 +41,16 @@ java -version
 # Initial command to trigger the execution of elasticsearch test 
 cd /elasticsearch
 
+set -e
 echo "Building elasticsearch  using gradlew \"gradlew assemble\"" && \
 ./gradlew -q -g /tmp assemble
 
-
-echo "Elasticsearch Build - Completed"
+echo "Elasticsearch Build - Successful"
 
 echo "Running elasticsearch tests :"
 
 ./gradlew -q -g /tmp test -Dtests.haltonfailure=false $TEST_OPTIONS
+
+echo "Elasticsearch tests - Successful"
+set +e
 find ./ -type d -name 'testJunit' -exec cp -r "{}" /testResults \;

